pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -406,9 +410,12 @@ public void addOrReplaceSegment(String segmentName, IndexLoadingConfig indexLoad
           localMetadata.getCrc(), zkMetadata.getCrc());
     }
     File indexDir = downloadSegment(segmentName, zkMetadata);
-    addSegment(indexDir, indexLoadingConfig);
-    LOGGER.info("Downloaded and loaded segment: {} of table: {} with crc: {}", segmentName, _tableNameWithType,
-        zkMetadata.getCrc());
+    Schema schema = ZKMetadataProvider.getTableSchema(_propertyStore, _tableNameWithType);
```

Comment:
We might want to remove the `addSegment(indexDir, indexLoadingConfig)` since it doesn't have all the required arguments, and can cause unexpected behavior when people use it

Reply:
mark it deprecated for now. I see it is only called in a method, which is not used any more.

Reply:
actually, we can change it as below so it loads segment to local tier by default, although this method is not used any more.
```
addSegment(ImmutableSegmentLoader.load(indexDir, indexLoadingConfig, schema, true, _tableDataDir, null));
```